caffe2/utils/math/reduce.h

```diff
@@ -5,8 +5,85 @@
 #include "caffe2/core/types.h"
 
 namespace caffe2 {
+
+class Tensor;
+
 namespace math {
 
+template <typename T, class Context>
+CAFFE2_API void
+ReduceMin(const int N, const T* X, T* y, Tensor* scratch_ptr, Context* context);
+
+template <typename T, class Context>
+CAFFE2_API void
+ReduceMax(const int N, const T* X, T* y, Tensor* scratch_ptr, Context* context);
+
+// Y = alpha * ReduceMin(X)
+template <typename T, class Context>
+CAFFE2_API void ReduceMin(
+    const int ndim,
```

Comment:
Explicitly mention that X_dims and Y_dims share the same ndim?

Reply:
Done